src/aspire/orientation/__init__.py

```diff
@@ -7,35 +15,434 @@ class Orient3D:
     """
     Define a base class for estimating 3D orientations
     """
-    def __init__(self, src):
+    def __init__(self, src, nrad=None, ntheta=None, masked=True):
         """
-        constructor of an object for estimating 3D orientations
+        Initialize an object for estimating 3D orientations
+
+        :param src: The source object of 2D denoised or class-averaged imag
+        :param nrad: The number of points in the radial direction
+        :param ntheta: The number of points in the theta direction
+        :param masked: Whether applying a mask to the images or not
         """
-        pass
+        self.src = src
+        self.nimg = src.n
+        self.nres = self.src.L
+        if nrad is None:
+            self.nrad = math.ceil(config.orient.r_ratio * self.nres)
+        else:
+            self.nrad = nrad
+        if ntheta is None:
+            self.ntheta = config.orient.n_theta
+        else:
+            self.ntheta = ntheta
+
+        self.basis = PolarBasis2D((self.nres, self.nres), self.nrad, self.ntheta)
+
+        self.max_shift = math.ceil(config.orient.max_shift * self.nres)
+        self.shift_step = config.orient.shift_step
+
+        imgs = self.src.images(start=0, num=np.inf).asnumpy()
```

Comment:
The number of input images here should come from the results after 2D classification. Probably a few thousands of class averages? Since the common lines matrix is in the size of n_images x n_images, probably we don't need to load images in batch size now. Othewise significant improvements need to be done for the rest of code. By the way, I will also develop a standard command for 3D orientation estimation after this PR.

Reply:
During the last meeting we decided to load all images first since we are no supposed to use all raw images in high resolution but classified 2D images in lower resolution. More significant improvements and redesignment are needed to calculate common lines, synchronization matrix, rotations, and shifts using batch size since operations on pairs of images have to be involved.  Those should be addressed in other PRs if really needed.

Reply:
Ok resolving this since we're going to keep this as a non-batched algorithm.